objective.py

```diff
@@ -6,54 +6,127 @@
-        self.X_train, self.y_train = X_train, y_train
-        self.X_test, self.y_test = X_test, y_test
+        elif self.evaluation_process == 'inter_subject':
+
+            sujet_test = self.subject_test
+            data_subject_test = data_split_subject[str(sujet_test)]
+            n_subject = len(data_split_subject)
+            data_subject_train = []
+            for i in range(1, n_subject+1):
+                if i != sujet_test:
+                    data_subject_train += data_split_subject[str(i)]
+
+            X_test = SliceDataset(data_subject_test, idx=0)
+            y_test = np.array([y for y in SliceDataset(data_subject_test,
+                                                       idx=1)])
+
+            X_train = SliceDataset(data_subject_train, idx=0)
+            y_train = np.array([y for y in SliceDataset(data_subject_train,
+                                                        idx=1)])
+
+            self.X_train, self.y_train = X_train,  y_train
+            self.X_test, self.y_test = X_test, y_test
+
+        elif self.evaluation_process == 'inter_session':
+
+            data_subject = data_split_subject[str(self.subject)]
+            data_split_session = data_subject.split('session')
+            session_test = self.session_test
+            data_session_test = data_split_session[session_test]
+            data_session_train = []
+            for key in data_split_session.items():
+                if key[0] != str(session_test):
+                    data_session_train += data_split_session[key[0]]
+            X_test = SliceDataset(data_session_test, idx=0)
+            y_test = np.array([y for y in SliceDataset(data_session_test,
+                                                       idx=1)])
+            X_train = SliceDataset(data_session_train, idx=0)
+            y_train = np.array([y for y in SliceDataset(data_session_train,
+                                                        idx=1)])
+
+            self.X_train, self.y_train = X_train, y_train
+            self.X_test, self.y_test = X_test, y_test
+
+        self.sfreq = sfreq
 
-        # The dictionary defines the keyword arguments for `Objective.set_data`
         return dict(
-            X_train=X_train, y_train=y_train,
-            X_test=X_test, y_test=y_test
+                X_train=X_train, y_train=y_train,
+                X_test=X_test, y_test=y_test,
+                sfreq=sfreq
         )
 
     def compute(self, model):
         # The arguments of this function are the outputs of the
         # `Solver.get_result`. This defines the benchmark's API to pass
         # solvers' result. This is customizable for each benchmark.
+        if not type(model) == 'braindecode.classifier.EEGClassifier':
+            self.X_train = to_numpy(self.X_train)
+            self.X_test = to_numpy(self.X_test)
+
         score_train = model.score(self.X_train, self.y_train)
         score_test = model.score(self.X_test, self.y_test)
```

Comment:
as we continue several metrics here, we should move to computing the test prediction once and then compute all the metrics like BAS.

Reply:
Sorry, this was the only point that I didn't catch